packages/storybook/src/schematics/migrate-defaults-5-to-6/migrate-defaults-5-to-6.ts

```diff
@@ -0,0 +1,286 @@
+          name: project,
+          uiFramework:
+            workspaceJson.projects[project].architect.storybook.options
+              .uiFramework,
+          configFolder:
+            workspaceJson.projects[project].architect.storybook.options.config
+              .configFolder,
+        };
+      }
+    });
+
+    return chain(
+      projectsThatHaveStorybookConfiguration
+        .filter(
+          (projectWithStorybook) =>
+            projectWithStorybook && projectWithStorybook.configFolder
+        )
+        .map((projectWithStorybook) => {
+          return chain([
+            migrateStorybookInstance(
+              keepOld,
+              projectWithStorybook.name,
+              projectWithStorybook.uiFramework,
+              projectWithStorybook.configFolder
+            ),
+          ]);
+        })
+    );
+  };
+}
+
+export function migrateStorybookInstance(
+  keepOld: boolean,
+  projectName: string,
+  uiFramework: string,
+  configFolder?: string
+): Rule {
+  return chain([
+    migrateProjectLevelStorybookInstance(
+      keepOld,
+      projectName,
+      uiFramework,
+      configFolder
+    ),
+  ]);
+}
+
+function maybeUpdateVersion(): Rule {
+  return updateJsonInTree('package.json', (json) => {
+    json.dependencies = json.dependencies || {};
+    json.devDependencies = json.devDependencies || {};
+
+    const allStorybookPackagesInDependencies = Object.keys(
+      json.dependencies
+    ).filter((packageName: string) => packageName.startsWith('@storybook/'));
+
+    const allStorybookPackagesInDevDependencies = Object.keys(
+      json.devDependencies
+    ).filter((packageName: string) => packageName.startsWith('@storybook/'));
+
+    const storybookPackages = [
+      ...allStorybookPackagesInDependencies,
+      ...allStorybookPackagesInDevDependencies,
+    ];
+
+    storybookPackages.forEach((storybookPackageName) => {
+      if (json.dependencies[storybookPackageName]) {
+        const version = checkAndCleanWithSemver(
+          storybookPackageName,
+          json.dependencies[storybookPackageName]
+        );
+        if (lte(version, '6.0.0')) {
+          json.dependencies[storybookPackageName] = storybookVersion;
+          needsInstall = true;
+        }
+      }
+      if (json.devDependencies[storybookPackageName]) {
+        const version = checkAndCleanWithSemver(
+          storybookPackageName,
+          json.devDependencies[storybookPackageName]
+        );
+        if (lte(version, '6.0.0')) {
+          json.devDependencies[storybookPackageName] = storybookVersion;
+          needsInstall = true;
+        }
+      }
+    });
+    return json;
+  });
+}
+
+function upgradeStorybookPackagesInPackageJson(): Rule {
+  /**
+   * Should upgrade all @storybook/* packages in package.json
+   */
+  return (tree: Tree, context: SchematicContext) => {
+    return chain([
+      maybeUpdateVersion(),
+      formatFiles(),
+      addInstallTask({ skipInstall: !needsInstall }),
```

Comment:
I checked and double and triple checked. In `maybeUpdateVersion()` the `needsInstall` gets turned to `true`. However, judging from the console logs, ` addInstallTask({ skipInstall: !needsInstall })` runs first, and `skipInstall` is `true` when it reaches that function. `maybeUpdateVersion()` runs afterwards. I do not really understand how this happens. Given that this code is the same as here: `packages/storybook/src/migrations/update-11-0-12/update-storybook.ts`

Reply:
When I manually pass `addInstallTask({ skipInstall: false })`, the installation happens, of course.

Reply:
These are the logs from the console, appearing in that order
```
# log inside addInstallTask function
OPTIONS OF ADD INSTALL { skipInstall: true }
# log inside maybeUpdateVersion function
true needsInstall # log 
UPDATE package.json
```